### PR TITLE
Rocket structure

### LINKED ISSUE #13: [4pt] Display rockets - Lists render
- Use `useSelector()` Redux Hook to select the state slices and render lists of rockets in corresponding routes. i.e.:
```javascript
// get rockets data from the store
const rockets = useSelector(state => state.rockets);
```

- You can style the whole application "by hand" or you could use [React Bootstrap](https://react-bootstrap.github.io/), a UI library that could speed up the process. This is a popular library and working with its components would be good practice.
- Render a list of rockets (as per design). For the image of a rocket use the first image in the array of `flickr_images`.

### LINKED ISSUE #17: [4pt]  Fetch rockets - Fetch data
Fetch data from the Rockets endpoint  (https://api.spacexdata.com/v3/rockets) when the application starts (as Rockets is the default view).

Once the data are fetched, dispatch an action to store the selected data in Redux store:

- id
- name
- type
- flickr_images

NOTE: Make sure you only dispatch those actions once and do not add data to store on every re-render (i.e. when changing views / using navigation).

### LINKED ISSUE #21: [1pt] Create basic structure for Rockets - Setup
- Create a route and a view component. Use `<NavLink />` for the page navigation links and style active class to indicate which section/page the user is currently on (underline active navigation link).
- Create a directory for all Redux state slice files.

### PR DESCRIPTION
**Pull request by davidigbo**
In this pull request, I implement fetching API, create rocket structure, and display list render functionality.

fixed #21 
fixed #13 
fixed #17

### CODE REVIEW

File: src/components/Rockets.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import { useEffect } from 'react';
+import { fetchRockets } from '../redux/rockets/rocketsSlice';
+
+const Rockets = () => {
+  const dispatch = useDispatch();
+  const rockets = useSelector((state) => state.rockets);
+  console.log(rockets);
+  useEffect(() => {
+    dispatch(fetchRockets());
+  }, [dispatch]);
```
Comment (evillaz) [inline src/components/Rockets.js:11]: Great work on integrating and calling the method to retrieve information from rockets. Although the requirements are that the rocket's information should be retrieved before the page first loads. Kindly move this block of code to App.js so we can ensure that the data is retrieved before the page loads

**Review by evillaz — APPROVED** [review]
## STATUS: APPROVED 👍🎯  
Hello @david30-maker 👋 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⭐ 
✔️ correct use of gitflow
✔️ linters are passing
✔️ professional documentation
## Optional suggestions

Cheers and Happy coding!👏👏👏

------

**Comment by davidigbo** [comment]
Thanks @evillaz,
I will implement the change.

File: src/components/Rockets.js (hunk 1 above)
Comment (davidigbo) [reply src/components/Rockets.js:11]: Hey @evillaz, please I don't understand what you meant here.
Do you think `useEffect` is allowed to be used in the App.js component?
You need to show me how to implement it in App.js. Thanks

**Pull request merged**